### PR TITLE
[Hawk alert] Additional fields and alert views

### LINKED ISSUE #7503: Establish additional fields for Hawk Alerts
# Story

As an Emergency site owner, I would like to be able to manually provide situation updates to Hawk Alerts and assign a status so that the community can stay informed throughout an Emergency.

# Additional Info

These fields are manually added/updated.

- Status - single valued - they want the ability to add new terms here. They only get output on the hawk alert node as text and don't get linked to. Rabbit hole. Want output higher up on hawk alert node rather than at bottom.
- Situation Updates - multiple - Paragraph collection with date and details similar to ITS alert updates.
  - want to output the date first on the hawk alert node rather than last like it is currently.
  - want this to be a little easier to read
  - would like the date to default to now rather than empty
  - list these chronologically with most recent updates first similar to what we did with ITS
- Complete - boolean to designate the alert is no longer active.

# Proposed Solution

### LINKED ISSUE #7520: Display active hawk alerts on emergency front page
# Story

As a site owner of emergency.uiowa.edu, I would like active hawk alerts to be easily visible at the top of the front page so that visitors can quickly get Emergency information or see that there are no current emergencies.

# Additional Info

If there are no active Hawk Alerts they want a message like they have currently Showing the campus status is normal and there are no emergencies at this time.

**Current text - https://emergency.uiowa.edu**

> Campus Status Normal
> There are no known emergencies at this time. The campus is operating normally.
> In the event of an emergency, this space will be used to provide timely information to the University community. It will be updated regularly as new information becomes available.

Any and all active hawk alerts should be displayed.

If more than one active hawk alert, they should be stacked and ordered with most recent first. Show all alert info including updates,  currently not linked to full node here.

# Proposed Solution
```[tasklist]
### Tasks
- [ ] Create a view to display active alerts.  Use field display and use alert classes.
```

### LINKED ISSUE #7521: recent hawk alerts and archive of all
# Story

As a site owner of emergency.uiowa.edu, I would like to display recent hawk alerts that are no longer active on our front page.

# Additional Info

Display 3-5 most recent hawk alerts that are no longer active with a link to view all past alerts.

Show title (linked to full node)
Show post date
Order by most recent first

They are checking on how many past hawk alerts they need to keep and how those should get added/imported

# Proposed Solution

### PR DESCRIPTION
**Pull request by bspeare**
Resolves https://github.com/uiowa/uiowa/issues/7503 
Resolves https://github.com/uiowa/uiowa/issues/7521
Resolves https://github.com/uiowa/uiowa/issues/7520

# How to test https://github.com/uiowa/uiowa/issues/7503 

```
 ddev blt frontend && ddev blt ds --site=emergency.uiowa.edu && ddev drush @emergency.local config:import --source ../config/sites/emergency.uiowa.edu --partial -y && ddev drush @emergency.local cim -y && ddev drush @emergency.local cr && ddev drush @emergency.local en devel_generate && ddev drush @emergency.local devel-generate:content 50 --bundles hawk_alert && ddev drush @emergency.local uli /node/add/hawk_alert
```

Confirm Status - is single valued 
  - Webmaster can add new terms here
  - They only get output on the hawk alert node as text and don't get linked to. Rabbit holed.
  - Displays higher up on hawk alert node rather than at bottom
    
Confirm Situation Updates 
  - Multiple - Paragraph collection with date and details similar to ITS alert updates
  - Outputs the date first on the hawk alert node rather than last like it is currently
  - Date defaults to now
  - Listed these chronologically with most recent updates first similar to what we did with ITS

Confirm Complete field
 - Boolean to designate the alert is no longer active
 - Does not display on node

# How to test  https://github.com/uiowa/uiowa/issues/7520
On https://emergency.uiowa.ddev.site/node/1/layout, add "Hawk alert - active" block to the layout.  Confirm it shows the most recent active hawk alerts first and situation update text in alert component.   Delete all active alerts and confirm that success style alert is showing with no results text. 

# How to test   https://github.com/uiowa/uiowa/issues/7521

- On https://emergency.uiowa.ddev.site/node/1/layout, add "Hawk alert - recent" block to the layout.  Confirm it shows only the last 5 closed alerts. 
- Go to https://emergency.uiowa.ddev.site/archive and confirm 10 items are showing with full pager with most recent alerts showing first. Active alerts should not show in this list.

### CODE REVIEW

File: config/sites/emergency.uiowa.edu/field.field.paragraph.hawk_alert_situation_updates.field_hawk_alert_situation_detai.yml
Comment (joewhitsitt) [inline config/sites/emergency.uiowa.edu/field.field.paragraph.hawk_alert_situation_updates.field_hawk_alert_situation_detai.yml:1]: too late to adjust the naming scheme for this to not be a cutoff? `detai`

Comment (bspeare) [reply config/sites/emergency.uiowa.edu/field.field.paragraph.hawk_alert_situation_updates.field_hawk_alert_situation_detai.yml:1]: Not too late :)

**Comment by joewhitsitt** [comment]
pulling to test and to resolve conflicts

**Comment by joewhitsitt** [comment]
merge conflicts were a bit messy. one thing that doesn't seem to be working is only the first delta of situation updates is displayed. in the view config, multiple are shown, so i think there is an issue with the render. i don't _think_ it got messed up with the merge. going to poke a bit more.

**Comment by joewhitsitt** [comment]
Looks like the view has the updates, but they are being pushed apart by a height 100:

Screenshot 2024-04-12 at 12 14 57 PM

**Comment by joewhitsitt** [comment]
```diff
+
+// Cards inside of alerts.
+.alert .card {
+  height: auto !important;
+}
```

This seems to wrangle it but not sure if we are doing cards in alerts anywhere else or what effect this has on cards overall.

**Comment by bspeare** [comment]
@joewhitsitt this is due to the vertical alignment of the section.   An issue like https://github.com/uiowa/uiowa/issues/6349 would resolve this for the platform.   For this particular alert with cards inside it, I think adjusting the container vertical alignment settings fixes the issue.  However, I have also added a style fix that will make the cards work even if the vertical alignment is not adjusted.

**Comment by joewhitsitt** [comment]
sounds good. i'll pull and test one more time. that was the only thing I think holding back approval.

**Review by joewhitsitt — APPROVED** [review]
Good work!

**Pull request merged**